api/testing-internal/src/main/java/io/opentelemetry/api/testing/internal/AbstractOpenTelemetryTest.java

```diff
@@ -99,14 +99,74 @@ void setThenSet() {
 
   @Test
   void getThenSet() {
-    assertThat(getGlobalOpenTelemetry().getClass().getName())
-        .isEqualTo("io.opentelemetry.api.DefaultOpenTelemetry");
+    assertThat(GlobalOpenTelemetry.getOrNoop()).isSameAs(OpenTelemetry.noop());
+    // Calling GlobalOpenTelemetry.get() has the side affect of setting GlobalOpenTelemetry to an
+    // (obfuscated) noop.
+    // Call GlobalOpenTelemetry.get() using a utility method so we can later assert it was
+    // responsible for setting GlobalOpenTelemetry.
+    assertThat(getGlobalOpenTelemetry())
+        .satisfies(
+            instance ->
+                assertThat(instance.getClass().getName())
+                    .isEqualTo("io.opentelemetry.api.GlobalOpenTelemetry$ObfuscatedOpenTelemetry"))
+        .extracting("delegate")
+        .isSameAs(OpenTelemetry.noop());
+    assertThat(GlobalOpenTelemetry.getOrNoop()).isNotSameAs(OpenTelemetry.noop());
     assertThatThrownBy(() -> GlobalOpenTelemetry.set(getOpenTelemetry()))
         .isInstanceOf(IllegalStateException.class)
         .hasMessageContaining("GlobalOpenTelemetry.set has already been called")
         .hasStackTraceContaining("getGlobalOpenTelemetry");
   }
 
+  private static OpenTelemetry getGlobalOpenTelemetry() {
+    return GlobalOpenTelemetry.get();
+  }
+
+  @Test
+  void getOrNoop() {
+    // Should be able to call getOrNoop multiple times without side effect and later call set.
+    assertThat(GlobalOpenTelemetry.getOrNoop()).isSameAs(OpenTelemetry.noop());
+    assertThat(GlobalOpenTelemetry.getOrNoop()).isSameAs(OpenTelemetry.noop());
+    setOpenTelemetry();
+    assertThat(GlobalOpenTelemetry.getOrNoop()).isNotSameAs(OpenTelemetry.noop());
```

Comment:
Sometimes obfuscated, sometime unobfuscated is a code smell https://github.com/open-telemetry/opentelemetry-java/pull/7819/files#r2529131045